src/components/settings/settings-knowledge-base/settings-knowledge-base.js

```diff
@@ -25,106 +28,125 @@ import NavigationModal from '../../navigation-modal';
 import ShowHidePasswordField from '../../show-hide-password-field';
 import { KbCredentials } from '../../../constants';
 
+const KB_NAME_LIMIT = 255;
 const focusOnErrors = createFocusDecorator();
-class SettingsKnowledgeBase extends Component {
-  static propTypes = {
-    config: KbCredentials.CredentialShape,
-    currentKBName: PropTypes.string.isRequired,
-    history: ReactRouterPropTypes.history.isRequired,
-    intl: PropTypes.object.isRequired,
-    isCreateMode: PropTypes.bool,
-    kbCredentials: KbCredentials.KbCredentialsReduxStateShape,
-    match: ReactRouterPropTypes.history.isRequired,
-    onDelete: PropTypes.func.isRequired,
-    onSubmit: PropTypes.func.isRequired,
-  };
-
-  state = {
-    toasts: [],
-    deleteConfirmationModalDisplayed: false,
-  }
-
-  componentDidUpdate(prevProps) {
-    const {
-      config,
-      kbCredentials,
-      history,
-    } = this.props;
 
+const propTypes = {
+  config: KbCredentials.CredentialShape,
+  currentKBName: PropTypes.string.isRequired,
+  isCreateMode: PropTypes.bool,
+  kbCredentials: KbCredentials.KbCredentialsReduxStateShape,
+  kbId: PropTypes.string.isRequired,
+  onDelete: PropTypes.func.isRequired,
+  onSubmit: PropTypes.func.isRequired,
+};
+
+const defaultProps = {
+  isCreateMode: false,
+};
+
+const SettingsKnowledgeBase = ({
+  config,
+  currentKBName,
+  isCreateMode,
+  kbCredentials,
+  kbId,
+  onDelete,
+  onSubmit,
+}) => {
+  const intl = useIntl();
+  const history = useHistory();
+
+  const [toasts, setToasts] = useState([]);
+  const [deleteConfirmationModalDisplayed, setDeleteConfirmationModalDisplayed] = useState(false);
+
+  useEffect(() => {
```

Comment:
I think it's better to split a single `useEffect` into several smaller ones, and instead of passing whole `kbCredentials` object as a dependency pass only `kbCredentials.hasUpdated`. This will guard against cases when `kbCredentials` reference has changed, but the property that we are checking is still the same, but code inside `if` will run twice anyway.